src/cache/engine/RedisCacheEngine.ts

```diff
@@ -25,7 +25,7 @@ class RedisCacheEngine implements ICacheEngine {
         const dateRegex = /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3}Z$/
         if (dateRegex.test(value)) {
           return new Date(value)
-        } else if (Types.ObjectId.isValid(value)) {
+        } else if (Types.ObjectId.isValid(value) && new Types.ObjectId(value).toString() === value) {
```

Comment:
do not build ObjectId twice: 
```typescript
else if (Types.ObjectId.isValid(value)) {
  const objectId = new Types.ObjectId(value)
  return objectId.toString() === value ? objectId : value
}
```

Reply:
This would always create an `ObjectID` as opposed to only when `Types.ObjectId.isValid(value)` renders `true`. In most situations, this would rather be less efficient.